samples/HostedInAspNet.Server/Startup.cs

```diff
@@ -9,21 +10,27 @@ namespace HostedInAspNet.Server
 {
     public class Startup
     {
-        // This method gets called by the runtime. Use this method to add services to the container.
-        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddBlazorCircuits();
+
+            services.AddHttpClient();
+            services.AddScoped<HttpClient>(s => s.GetRequiredService<IHttpClientFactory>().CreateClient());
         }
 
-        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
         public void Configure(IApplicationBuilder app, IHostingEnvironment env)
         {
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
             }
 
-            app.UseBlazor<Client.Program>();
+            app.UseBlazorOnTheServer(renderer =>
```

Comment:
`UseServerSideBlazor`?

Reply:
I expect you've considered this, but what about the possibility of changing this to just:

    app.UseServerSideBlazor<Client.Startup>();

... so that the nomination of root components is contained in the Blazor app code?

Also I'm not sure where the developer would configure per-circuit DI services, such as an `AppState` singleton. Do you envision this going in `ConfigureServices` in the server project? Is it relatively simple to scope something to a circuit?

If it was possible to have `app.UseServerSideBlazor<Client.Startup>();` here and then configure per-circuit DI services in the `ConfigureServices` method of the `Startup` class in the Blazor client lib, that would be extremely clear I think and would also make it simpler for developers to try switching from server-executed to WebAssembly-executed hosting models early in their project (until such time as they strongly take dependency on other aspects of a chosen hosting model).

Reply:
I'm really confused by this feedback, this is opposite of the discussion we had that lead to https://github.com/aspnet/Blazor/commit/128b38fce62bd8d495b7993099cb39aced601b60


-- 
> In both cases these would invoke the corresponding methods on Client.Startup (i.e., ConfigureServices and Configure) so that the developer can configure DI services there if they want. This would be useful if they wanted, for example, to register a scoped AppState when that type is internal to the Blazor library project.

In this case where is the service provider that's being configured? Is it the main ASP.NET Core service provider?

Reply:
Actually I missed something here. This doesn't look like anything else we've done but we could try it.

Reply:
I'm unsure which specific bit of discussion it's the opposite to (having just re-read the comments on #1092), but I fully recognise I might be forgetting something we covered verbally!

> In this case where is the service provider that's being configured? Is it the main ASP.NET Core service provider?

Yes, I think we're proposing to have only one service provider.